sbc_compassion/models/correspondence_template.py

```diff
@@ -59,31 +62,35 @@ class CorrespondenceTemplate(models.Model):
     page_height = fields.Integer(
         help='Height of the template in pixels')
     bluesquare_x = fields.Integer(
-        readonly=True,
+        compute='_compute_template_keypoints', store=True,
         help='X Position of the upper-right corner of the bluesquare '
              'in pixels')
     bluesquare_y = fields.Integer(
-        readonly=True,
+        compute='_compute_template_keypoints', store=True,
         help='Y Position of the upper-right corner of the bluesquare '
              'in pixels')
     qrcode_x_min = fields.Integer(
+        compute="_onchange_template_image",
         help='Minimum X position of the area in which to look for the QR '
              'code inside the template (given in pixels)')
     qrcode_x_max = fields.Integer(
+        compute="_onchange_template_image",
         help='Maximum X position of the area in which to look for the QR '
              'code inside the template (given in pixels)')
     qrcode_y_min = fields.Integer(
+        compute="_onchange_template_image",
         help='Minimum Y position of the area in which to look for the QR '
              'code inside the template (given in pixels)')
     qrcode_y_max = fields.Integer(
+        compute="_onchange_template_image",
         help='Maximum Y position of the area in which to look for the QR '
              'code inside the template (given in pixels)')
     pattern_center_x = fields.Float(
-        readonly=True,
+        compute="_compute_template_keypoints",
         help='X coordinate of the center of the pattern. '
         'Used to detect the orientation of the pattern in the image')
     pattern_center_y = fields.Float(
-        readonly=True,
+        compute="_compute_template_keypoints",
```

Comment:
Why is bluesquare stored but not pattern_center ? If pattern is not stored, `_compute_template_keypoints` will be called each time a template is opened.